packages/palette/src/elements/Pill/Pill.tsx

```diff
@@ -0,0 +1,159 @@
+import { themeGet } from "@styled-system/theme-get"
+import React from "react"
+import styled from "styled-components"
+import { css } from "styled-components"
+import { variant } from "styled-system"
+import { DROP_SHADOW, INNER_SHADOW } from "../../helpers/shadow"
+import { CheckIcon } from "../../svgs"
+import { BoxProps } from "../Box"
+import { Clickable, ClickableProps } from "../Clickable"
+import { Text } from "../Text"
+
+export const PILL_VARIANT_NAMES = [
+  "textRound",
+  "textSquare",
+  "filter",
+  "artist",
+] as const
+
+type PillVariant = typeof PILL_VARIANT_NAMES[number]
+type PillState = "focus" | "hover" | "active"
+
+const STATES: Record<PillState, any> = {
+  hover: css`
+    background-color: transparent;
+    border-color: transparent;
+    box-shadow: ${DROP_SHADOW};
+  `,
+  focus: css`
+    background-color: ${themeGet("colors.black10")};
+    border-color: transparent;
+    box-shadow: ${INNER_SHADOW};
+  `,
+  active: css`
+    background-color: transparent;
+    border-color: ${themeGet("colors.black60")};
+  `,
+}
+
+const PILL_VARIANTS: Record<PillVariant, BoxProps> = {
+  textRound: {
+    borderRadius: "15px",
+    height: "30px",
+    px: 1,
+  },
+  textSquare: {
+    height: "30px",
+    px: 1,
+  },
+  filter: {
+    borderRadius: "20px",
+    height: "40px",
+    px: 2,
+  },
+  artist: {
+    borderRadius: "25px",
+    height: "50px",
+    pr: 2,
+    pl: 1,
+  },
+}
+
+/** PillProps */
+export type PillProps = ClickableProps & {
+  as?: keyof JSX.IntrinsicElements | React.ComponentType
+  /** Forces focus state */
+  focus?: boolean
+  /** Forces hover state */
+  hover?: boolean
+} & (
+    | {
+        variant?: Extract<PillVariant, "textRound" | "textSquare" | "filter">
+        /** Forces active state */
+        active?: boolean
+      }
+    | {
+        /** `"artist"` pills do not have an `active` state */
+        variant?: Extract<PillVariant, "artist">
+        /**
+         * Optional avatar; [1x] or [1x, 2x]
+         * Should target 30x30 @1x, 60x60 @2x
+         */
+        srcs?: [string] | [string, string]
```

Comment:
Thoughts on having a little less friction here?
```
src: string | [string, string]
```
Then one can pass a simple string or an src set, and internally can handle it below.

Reply:
Yeah, good idea